src/Workspaces/Core/Portable/CodeCleanup/AbstractCodeCleanerService.cs

```diff
@@ -452,14 +458,112 @@ private bool CleanupWholeNode(TextSpan nodeSpan, ImmutableArray<TextSpan> spans)
             return spans[0].Contains(nodeSpan);
         }
 
-        private async Task<Document> IterateAllCodeCleanupProvidersAsync(
+        private async Task<Document> RunAllCodeCleanupProvidersInParallelAsync(
+            Document originalDocument,
+            ImmutableArray<ICodeCleanupProvider> codeCleaners,
+            CancellationToken cancellationToken)
+        {
+            using (Logger.LogBlock(FunctionId.CodeCleanup_RunAllCodeCleanupProvidersInParallel, cancellationToken))
+            {
+#if DEBUG
+                bool originalDocHasErrors = await originalDocument.HasAnyErrorsAsync(cancellationToken).ConfigureAwait(false);
+#endif
+                
+                // Run all the cleaners in parallel.  Keeping what results we can if they don't
+                // conflict with others.
+                var remainingCleaners = codeCleaners.ToList();
+                var iterationCount = 0;
+
+                var currentDocument = originalDocument;
+                while (remainingCleaners.Count > 0)
+                {
+                    iterationCount++;
```

Comment:
@heejaechang @sharwell I would like to report telemetry on this variable.  So we can get a sense of how many loops we're generally having to take until we complete all providers.  But i couldn't figure out how to report back a numeric value associated with some ID.  Could you let me know how to do that?

Reply:
this is how you do it

http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Diagnostics/Log/DiagnosticAnalyzerLogger.cs,69

add new FunctionId and do Logger.Log(FucntionId.XXXX, KeyValueLogMessage.Create(m => m["nameOfProperty"] = value);